src/Raven.Client/Documents/Operations/ETL/Queue/AzureQueueStorageConnectionSettings.cs

```diff
@@ -0,0 +1,163 @@
+        return $"https://{storageAccountName}.queue.core.windows.net/";
+    }
+    
+    private string GetUrlFromConnectionString(string connectionString)
+    {
+        var parts = connectionString.Split(';')
+            .Select(p => p.Split('='))
+            .ToDictionary(p => p[0], p => p.Length > 1 ? p[1] : "");
+        
+        parts.TryGetValue("DefaultEndpointsProtocol", out string protocol);
+        if (string.IsNullOrWhiteSpace(protocol))
+        {
+            HandleConnectionStringError("Protocol not found in the connection string");
+        }
+
+        if (protocol.Equals("http"))
+        {
+            parts.TryGetValue("QueueEndpoint", out string queueEndpoint);
+            if (string.IsNullOrWhiteSpace(queueEndpoint))
+            {
+                HandleConnectionStringError("Queue endpoint not found in the connection string");
+            }
+
+            return queueEndpoint;
+        }
+
+        parts.TryGetValue("AccountName", out string accountName);
+        if (string.IsNullOrWhiteSpace(accountName))
+        {
+            HandleConnectionStringError("Storage account name not found in the connection string");
+        }
+        
+        return $"https://{accountName}.queue.core.windows.net/";
+    }
+
+    private string GetStorageAccountName()
+    {
+        string storageAccountName = "";
+        
+        if (EntraId != null)
+        {
+            storageAccountName = EntraId.StorageAccountName;
+        }
+        else if (Passwordless != null)
+        {
+            storageAccountName = Passwordless.StorageAccountName;
+        }
+
+        return storageAccountName;
+    }
+    
+    private void HandleConnectionStringError(string message)
+    {
+        throw new ArgumentException(message, nameof(ConnectionString));
+    }
+
+    public DynamicJsonValue ToJson()
+    {
+        var json = new DynamicJsonValue
+        {
+            [nameof(ConnectionString)] = ConnectionString,
+            [nameof(EntraId)] = EntraId == null
+                ? null
+                : new DynamicJsonValue
+                {
+                    [nameof(EntraId.StorageAccountName)] = EntraId?.StorageAccountName,
+                    [nameof(EntraId.TenantId)] = EntraId?.TenantId,
+                    [nameof(EntraId.ClientId)] = EntraId?.ClientId,
+                    [nameof(EntraId.ClientSecret)] = EntraId?.ClientSecret
+                },
+            [nameof(Passwordless)] = Passwordless == null
+                ? null
+                : new DynamicJsonValue { [nameof(Passwordless.StorageAccountName)] = Passwordless?.StorageAccountName }
+        };
+
+        return json;
+    }
+}
+
+public sealed class EntraId
+{
+    public string StorageAccountName { get; set; }
+    public string TenantId { get; set; }
+    public string ClientId { get; set; }
+    public string ClientSecret { get; set; }
+
+    public bool IsValid()
+    {
+        return !string.IsNullOrWhiteSpace(StorageAccountName) &&
```

Comment:
```suggestion
        return string.IsNullOrWhiteSpace(StorageAccountName) == false &&
```
improves readability, same goes for lines below

Reply:
done